logstash-core/lib/logstash/instrument/periodic_poller/dlq.rb

```diff
@@ -10,13 +10,11 @@ def initialize(metric, agent, options = {})
     end
 
     def collect
-      pipelines = @agent.with_running_user_defined_pipelines {|pipelines| pipelines}
-      unless pipelines.nil?
-        pipelines.each {|_, pipeline|
-          unless pipeline.nil?
-            pipeline.collect_dlq_stats
-          end
-        }
+      pipelines = @agent.running_user_defined_pipelines
+      pipelines.each do |_, pipeline|
+        unless pipeline.nil?
```

Comment:
In what scenarios could we have a `PipelineState` that is running, but for which we have no `Pipeline`? Can we guard against this happening upstream to simplify usage?

From what I can tell in the code, this is _theoretically_ possible if the block passed to `PipelineRegistry#reload_pipeline` failed to emit a new `Pipeline` in its tuple, but that the only use of `PipelineRegistry#reload_pipeline` does not do this.

Reply:
good point, I refactored this code by essentially swapping the `with_running_user_defined_pipelines ` with `running_user_defined_pipelines` without really analyzing the original behaviour; this check is actually not necessary and we should never have a `nil` pipeline, not event from reload. From a defensive POV I am a little hesitant to remove that check to avoid unseen corner cases - I suggest to add this as a followup refactor PR to make sure there are no corner cases.

Reply:
for reference; added followup issue #10405